lib/chef/provider/remote_directory.rb

```diff
@@ -67,13 +67,10 @@ def ls(path)
                  ::File::FNM_DOTMATCH)
 
         # Remove current directory and previous directory
-        files.reject! do |name|
+        files = files.reject do |name|
```

Comment:
you don't think you need `files =` as it will be returned since it's the last expression in the method